python/gui/MaaS/cbv/DatasetApiView.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+from django.http import JsonResponse
+from wsgiref.util import FileWrapper
+from django.http.response import StreamingHttpResponse
+from .AbstractDatasetView import AbstractDatasetView
+from .DatasetFileWebsocketFilelike import DatasetFileWebsocketFilelike
+import logging
+logger = logging.getLogger("gui_log")
+
+
+class DatasetApiView(AbstractDatasetView):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _get_dataset_content_details(self, dataset_name: str):
+        result = asyncio.get_event_loop().run_until_complete(self.dataset_client.get_dataset_content_details(name=dataset_name))
```

Comment:
Looks like there are several calls to `asyncio.get_event_loop().run_until_complete`. It might help to have some synchronous versions of these functions on the dataset client since there appears to be some intention to call them synchronously. Maybe.

Reply:
The base functions for sending data through websockets are (and I am pretty sure must be) `async`, as are the functions making sure the client is authenticated (which relies on the former also).   It would take a decent bit of changing to be able to do this and have everything work, but we can look more at that in the future.